TradeItIosTicketSDK2/TradeItOrdersViewController.swift

```diff
@@ -0,0 +1,94 @@
+import UIKit
+import PromiseKit
+
+class TradeItOrdersViewController: TradeItViewController, TradeItOrdersTableDelegate {
+    
+    var ordersTableViewManager: TradeItOrdersTableViewManager?
+    let alertManager = TradeItAlertManager()
+    let tradingUIFlow = TradeItTradingUIFlow()
+    
+    var linkedBrokerAccount: TradeItLinkedBrokerAccount?
+    
+    @IBOutlet weak var ordersTable: UITableView!
+    @IBOutlet var orderTableBackgroundView: UIView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        guard let _ = self.linkedBrokerAccount else {
+            preconditionFailure("TradeItIosTicketSDK ERROR: TradeItOrdersViewController loaded without setting linkedBrokerAccount.")
+        }
+        self.ordersTableViewManager = TradeItOrdersTableViewManager(noResultsBackgroundView: orderTableBackgroundView)
+        self.ordersTableViewManager?.delegate = self
+        self.ordersTableViewManager?.ordersTable = self.ordersTable
+        
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        self.ordersTableViewManager?.initiateRefresh()
+    }
+    
+    //MARK: IBAction
+    
+    @IBAction func tradeButtonWasTapped(_ sender: Any) {
+        let order = TradeItOrder(linkedBrokerAccount: linkedBrokerAccount)
+        self.tradingUIFlow.presentTradingFlow(fromViewController: self, withOrder: order)
+    }
+    
+    // MARK: TradeItOrdersTableDelegate
+    
+    func refreshRequested(onRefreshComplete: @escaping () -> Void) {
+        guard let linkedBrokerAccount = self.linkedBrokerAccount
+            , let linkedBroker = linkedBrokerAccount.linkedBroker else {
+                preconditionFailure("TradeItIosTicketSDK ERROR: TradeItOrdersViewController loaded without setting linkedBrokerAccount.")
+        }
+        func authenticatePromise() -> Promise<Void>{
+            return Promise<Void> { fulfill, reject in
+                linkedBroker.authenticateIfNeeded(
+                    onSuccess: fulfill,
+                    onSecurityQuestion: { securityQuestion, answerSecurityQuestion, cancelSecurityQuestion in
+                        self.alertManager.promptUserToAnswerSecurityQuestion(
+                            securityQuestion,
+                            onViewController: self,
+                            onAnswerSecurityQuestion: answerSecurityQuestion,
+                            onCancelSecurityQuestion: cancelSecurityQuestion
+                        )
+                    },
+                    onFailure: reject
+                )
+            }
+        }
+        
+        func ordersPromise() -> Promise<Void> {
+            return Promise<Void> { fulfill, reject in
+                linkedBrokerAccount.getAllOrderStatus(
+                    onSuccess: { orders in
+                        self.ordersTableViewManager?.updateOrders(orders)
```

Comment:
Because this is a side effect rather than the request for data, it might be best to move this below in to its own `then` so set this to `onSuccess: fulfill` and change the promise type to accept the orders. Then below add:
```swift
authenticatePromise().then { _ in
    return ordersPromise()
}.then { orders in
    self.ordersTableViewManager?.updateOrders(orders)
}.catch //... etc
```

Reply:
I realise I didn't do this in the recent changes I made to portfolio. I'll try fix it there.